packages/form-upload/Upload.js

```diff
@@ -37,12 +54,74 @@ const Upload = ({
 
   const fieldValue = Array.isArray(field.value) ? field.value : [];
 
+  const callFileDelivery = useCallback(
+    async upload => {
+      if (!Array.isArray(upload)) upload = [upload];
+      const uploadResults = [];
+      try {
+        for (const u of upload) {
+          const data = {
+            deliveries: [
+              {
+                deliveryChannel,
+                fileURI: u.references[0],
+                metadata:
+                  typeof fileDeliveryMetadata === 'function'
+                    ? fileDeliveryMetadata(u)
```

Comment:
we should probably await this function in case what gets passed in is async. thoughts?

Reply:
Ah I hadn't thought of that, so the intended use case for when `fileDeliveryMetadata` is a function is for extracting props from the `upload` and populating specific/unique props that might be needed downstream. 

On one hand I think if something async or more complicated is needed then a dev would just use their own `onFileUpload` function anyway, on the other hand it's not hard to await this. @strange1120  what do you think?